WalletWasabi/Wallets/Wallet.cs

```diff
@@ -191,25 +191,7 @@ public List<TransactionSummary> BuildHistorySummary()
 	/// Gets the wallet transaction with the given txid, if the transaction exists.
 	/// </summary>
 	public bool TryGetTransaction(uint256 txid, [NotNullWhen(true)] out SmartTransaction? smartTransaction)
-	{
-		foreach (SmartCoin coin in GetAllCoins())
-		{
-			if (coin.TransactionId == txid)
-			{
-				smartTransaction = coin.Transaction;
-				return true;
-			}
-
-			if (coin.SpenderTransaction is not null && coin.SpenderTransaction.GetHash() == txid)
-			{
-				smartTransaction = coin.SpenderTransaction;
-				return true;
-			}
-		}
-
-		smartTransaction = null;
-		return false;
-	}
+		=> BitcoinStore.TransactionStore.TryGetTransaction(txid, out smartTransaction);
```

Comment:
These transactions can be related to any wallet.
But we could access `Coins.CoinsByTransactionId.Keys` and then ask to the transaction store if the TX is linked to our wallet

Reply:
Sorry, you just have to call `IsKnown(uint256 txid)`

Reply:
Whether I'm missing something, whether`IsKnown` is for all transactions, `Spent` and `Unspent`. You can see in the CoinsRegistry that Spending a coin does nothing on this cache.